### PR TITLE
🔥 1070 adding trigger tab for test page

### PR DESCRIPTION
**Pull request by xoscar**
This PR adds the trigger tab to the test run page.

## Changes

- Adding edit form to the trigger view.
- Adds response body and headers.
- Updates edit sequence to trigger the save and rerun.,

## Fixes

- https://github.com/kubeshop/tracetest/issues/1070

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [x] added a test

https://www.loom.com/share/9f2d2a60f5e2424b8da07653d4f4d341

### CODE REVIEW

File: web/cypress/integration/Home/EditTest.spec.ts [hunk 1]
```diff
@@ -7,19 +7,8 @@ describe('Edit Test', () => {
...
       const testId = getTestId(pathname);
-      cy.visit(`/`);
       cy.wait('@testList');
       cy.editTestByTestId(testId);
     });
   });
-
-  it('should edit a test from the test details', () => {
-    cy.location('pathname').then(pathname => {
-      const testId = getTestId(pathname);
```
Comment (xoscar) [inline web/cypress/integration/Home/EditTest.spec.ts:18]: removing unnecessary test as the only way to access the edit test is by selecting a run

File: web/cypress/support/commands.ts [hunk 2]
```diff
@@ -141,12 +145,14 @@ Cypress.Commands.add('createTestByName', (name: string) => {
...
 
-Cypress.Commands.add('editTestByTestId', (testId: string) => {
-  cy.get(`[data-cy=test-actions-button-${testId}]`).click();
-  cy.get('[data-cy=test-card-edit]').click();
+Cypress.Commands.add('editTestByTestId', () => {
+  cy.interceptEditTestCall();
   cy.get('[data-cy=edit-test-form]').should('be.visible');
   cy.get('[data-cy=create-test-name-input] input').clear().type('Edited Test');
   cy.get('[data-cy=edit-test-submit]').click();
+  cy.wait('@testEdit');
```
Comment (xoscar) [inline web/cypress/support/commands.ts:153]: Waiting for test edit and fetching the run

File: web/src/components/CreateTestPlugins/Grpc/steps/RequestDetails/RequestDetailsForm.tsx [hunk 3]
```diff
@@ -24,50 +23,52 @@ const RequestDetailsForm = ({form}: IProps) => {
...
-                {methodList.map(method => (
-                  <Select.Option data-cy={`rpc-method-${method}`} key={method} value={method}>
-                    {method}
-                  </Select.Option>
-                ))}
-              </Select>
-            </Form.Item>
           </span>
         </Col>
+        <Col span={12}>
```
Comment (xoscar) [inline web/src/components/CreateTestPlugins/Grpc/steps/RequestDetails/RequestDetailsForm.tsx:43]: Updating it so it is a single column form

File: web/src/components/EditTestForm/EditTestForm.tsx [hunk 4]
```diff
@@ -23,6 +25,16 @@ const EditTestForm = ({
   },
   onValidation,
 }: IProps) => {
+  const initialValues = useMemo(
```
Comment (xoscar) [inline web/src/components/EditTestForm/EditTestForm.tsx:28]: New way for initial values, using a factory function from the TestService

File: web/src/components/Router/Router.tsx [hunk 5]
```diff
@@ -6,15 +6,13 @@ import Home from 'pages/Home';
...
 import CreateTest from 'pages/CreateTest';
-import EditTest from 'pages/EditTest';
 
 const {serverPathPrefix = '/'} = window.ENV || {};
 
 const Router = () => (
   <HistoryRouter history={history} basename={serverPathPrefix}>
     <Routes>
       <Route path="/" element={<Home />} />
-      <Route path="/test/:testId/edit" element={<EditTest />} />
```
Comment (xoscar) [inline web/src/components/Router/Router.tsx:17]: Bye bye edit test page

File: web/src/components/RunDetailTrigger/RunDetailTrigger.tsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+const RunDetailTrigger = ({test, test: {id}, run: {triggerResult, state, executionTime}, run, isError}: IProps) => {
+  const shouldDisplayError = isError || state === TestState.FAILED;
+
+  return (
+    <S.Container>
+      <S.SectionLeft>
+        <EditTest test={test} />
+      </S.SectionLeft>
+      <S.SectionRight>
+        {shouldDisplayError ? (
```
Comment (xoscar) [inline web/src/components/RunDetailTrigger/RunDetailTrigger.tsx:24]: Showing the error prompt at the result level so the test can be edited and reran

File: web/src/components/RunDetailTriggerResponse/RunDetailTriggerResponse.tsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+        <div>
+          <S.StatusText>
+            Status: <S.StatusSpan $isError={statusCode >= 400}>{statusCode}</S.StatusSpan>
+          </S.StatusText>
+          <S.StatusText>
+            Time: <S.StatusSpan $isError={executionTime > 1000}>{executionTime}ms</S.StatusSpan>
+          </S.StatusText>
+        </div>
+      </S.TitleContainer>
+      <S.TabsContainer>
```
Comment (xoscar) [inline web/src/components/RunDetailTriggerResponse/RunDetailTriggerResponse.tsx:38]: Result + headers section

File: web/src/models/TestRun.model.ts [hunk 8]
```diff
@@ -53,7 +54,7 @@ const TestRun = ({
     executionTime,
     lastErrorState,
     trigger,
-    triggerResult,
+    triggerResult: rawTriggerResult ? TriggerResult(rawTriggerResult) : undefined,
```
Comment (xoscar) [inline web/src/models/TestRun.model.ts:57]: Models for trigger results

File: web/src/services/Test.service.ts [hunk 9]
```diff
@@ -49,8 +55,8 @@ const TestService = () => ({
       },
       ...(original
         ? {
-            definition: {
-              definitions: original.definition.definitionList.map(def => TestDefinitionService.toRaw(def)),
```
Comment (xoscar) [inline web/src/services/Test.service.ts:53]: Fixing a 🐛

File: web/cypress/support/commands.ts (hunk 2 above)
Comment (cescoferraro) [reply web/cypress/support/commands.ts:153]: 🔥

File: web/src/components/RunDetailTrigger/RunDetailTrigger.styled.ts [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+export const Section = styled.div`
+  flex: 1;
+`;
+
+export const SectionLeft = styled(Section)`
+  background-color: ${({theme}) => theme.color.white};
+  z-index: 1;
+`;
+
```
Comment (jorgeepc) [inline web/src/components/RunDetailTrigger/RunDetailTrigger.styled.ts:17]: Do we need to add scroll here too? I saw your video and it seems that the scroll is applied to the page and not the sections.

**Review by jorgeepc — APPROVED** [review]
Looks good man! One minor question

**Review by cescoferraro — APPROVED** [review]
Great job man! Could not find anything to criticize. LGreatTM

**Pull request merged**